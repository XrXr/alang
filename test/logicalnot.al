main :: proc() {
	a := false
	if !a {
		puts("1/4 success\n")
	}
	if !!a {
	} else {
		puts("2/4 success\n")
	}
	b := true
	if !b {
	} else {
		puts("3/4 success\n")
	}

	c := &a
	if !c {
	} else {
		puts("4/4 success\n")
	}
	// TODO add a testcase here when we zero out decls
}
