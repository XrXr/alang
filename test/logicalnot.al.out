1/4 success
2/4 success
3/4 success
4/4 success
